File: Helpers/RdpScreenSizes.cs
using System;
using System.Collections.Generic;

namespace Wormhole.Helpers;

/// <summary>
/// Shared RDP screen-size constants. The editor's screen-size picker and the ActiveX host's
/// resolver both look at the same strings — if the sentinel was duplicated as a literal
/// (and one got renamed for localisation), the resolver would silently fall back to its
/// fixed-size default instead of fitting the embedded RDP surface.
/// </summary>
public static class RdpScreenSizes
{
    /// <summary>Canonical editor value meaning "size the remote desktop to the available
    /// embedded connection content area".</summary>
    public const string FullConnectionContent = "Full connection content";

    /// <summary>Legacy value saved by older Wormhole builds for the same dynamic sizing mode.</summary>
    public const string LegacyFullScreenSentinel = "Full screen";

    /// <summary>Legacy value imported from mRemoteNG for the same dynamic sizing mode.</summary>
    public const string MRemoteNgFitToWindowSentinel = "FitToWindow";

    /// <summary>Back-compat alias for existing callers/tests that still refer to the old constant name.</summary>
    [Obsolete("Use FullConnectionContent for new UI and IsFullConnectionContent for policy checks.")]
    public const string FullScreenSentinel = LegacyFullScreenSentinel;

    /// <summary>Preset list surfaced by the editor's screen-size combo. Mirrors the mstsc
    /// resolution presets, with the content-sized dynamic mode first.</summary>
    public static IReadOnlyList<string> Presets { get; } = new[]
    {
        FullConnectionContent,
        "640x480", "800x600", "1024x768", "1280x800", "1280x1024",
        "1366x768", "1440x900", "1600x900", "1680x1050", "1920x1080",
    };

    /// <summary>
    /// True when the requested screen size should track the embedded connection surface instead
    /// of pinning a fixed remote desktop resolution. Null/empty keeps the historical default,
    /// old saved "Full screen" rows keep working, and mRemoteNG "FitToWindow" imports get the
    /// same dynamic remote-resolution behavior instead of a scaled fixed fallback.
    /// </summary>
    public static bool IsFullConnectionContent(string? screenSize) =>
        string.IsNullOrWhiteSpace(screenSize) ||
        string.Equals(screenSize, FullConnectionContent, StringComparison.OrdinalIgnoreCase) ||
        string.Equals(screenSize, LegacyFullScreenSentinel, StringComparison.OrdinalIgnoreCase) ||
        string.Equals(screenSize, MRemoteNgFitToWindowSentinel, StringComparison.OrdinalIgnoreCase);

    /// <summary>Map saved/imported dynamic aliases onto the canonical picker item while
    /// preserving null as "inherit/default".</summary>
    public static string? NormalizeForPicker(string? screenSize)
    {
        if (string.IsNullOrWhiteSpace(screenSize)) return null;
        return IsFullConnectionContent(screenSize) ? FullConnectionContent : screenSize;
    }
}

File: rust/crates/wormhole-domain/src/rdp_screen_sizes.rs
/// Shared RDP screen-size constants (`Wormhole.Helpers.RdpScreenSizes`).
pub struct RdpScreenSizes;

impl RdpScreenSizes {
    /// Canonical editor value: size remote desktop to the embedded content area.
    pub const FULL_CONNECTION_CONTENT: &'static str = "Full connection content";

    /// Legacy value from older Wormhole builds.
    pub const LEGACY_FULL_SCREEN_SENTINEL: &'static str = "Full screen";

    /// Legacy mRemoteNG import sentinel.
    pub const M_REMOTE_NG_FIT_TO_WINDOW_SENTINEL: &'static str = "FitToWindow";

    pub const PRESETS: &'static [&'static str] = &[
        Self::FULL_CONNECTION_CONTENT,
        "640x480",
        "800x600",
        "1024x768",
        "1280x800",
        "1280x1024",
        "1366x768",
        "1440x900",
        "1600x900",
        "1680x1050",
        "1920x1080",
    ];

    pub fn is_full_connection_content(screen_size: Option<&str>) -> bool {
        match screen_size {
            None => true,
            Some(s) if s.trim().is_empty() => true,
            Some(s) => {
                eq_ignore_ascii_case(s, Self::FULL_CONNECTION_CONTENT)
                    || eq_ignore_ascii_case(s, Self::LEGACY_FULL_SCREEN_SENTINEL)
                    || eq_ignore_ascii_case(s, Self::M_REMOTE_NG_FIT_TO_WINDOW_SENTINEL)
            }
        }
    }

    pub fn normalize_for_picker(screen_size: Option<&str>) -> Option<String> {
        match screen_size {
            None => None,
            Some(s) if s.trim().is_empty() => None,
            Some(s) if Self::is_full_connection_content(Some(s)) => {
                Some(Self::FULL_CONNECTION_CONTENT.to_string())
            }
            Some(s) => Some(s.to_string()),
        }
    }
}

fn eq_ignore_ascii_case(a: &str, b: &str) -> bool {
    a.eq_ignore_ascii_case(b)
}
